src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php

```diff
@@ -0,0 +1,667 @@
+            '------------------------------',
+            '<info>Webspace: </info>' . $this->webspaceKeySource . ' => ' . $this->webspaceKeyDestination,
+            '<info>Locale: </info>' . $localeSource . ' => ' . $localeDestination,
+            '------------------------------',
+        ]);
+
+        /** @var HomeDocument $homeDocumentSource */
+        $homeDocumentSource = $this->documentManager->find(
+            $this->sessionManager->getContentPath($this->webspaceKeySource),
+            $localeSource
+        );
+
+        // Generate all needed page documents.
+        $this->recursiveCopy(
+            $homeDocumentSource,
+            null,
+            $localeDestination
+        );
+
+        $this->output->writeln([
+            '------------------------------',
+            '',
+        ]);
+    }
+
+    /**
+     * Copies a given webspace with given locale to a destination webspace with a destination locale.
+     *
+     * @param string $localeSource
+     * @param string $localeDestination
+     */
+    protected function copyRedirectsAndStructure(
+        $localeSource,
+        $localeDestination
+    ) {
+        $this->output->writeln([
+            '------------------------------',
+            '<info>Webspace: </info>' . $this->webspaceKeySource . ' => ' . $this->webspaceKeyDestination,
+            '<info>Locale: </info>' . $localeSource . ' => ' . $localeDestination,
+            '------------------------------',
+        ]);
+
+        /** @var HomeDocument $homeDocumentSource */
+        $homeDocumentSource = $this->documentManager->find(
+            $this->sessionManager->getContentPath($this->webspaceKeySource),
+            $localeSource
+        );
+
+        // Generate redirect and structure.
+        $this->recursiveCopyRedirectsAndStructure(
+            $homeDocumentSource,
+            $localeDestination
+        );
+
+        $this->output->writeln([
+            '------------------------------',
+            '',
+        ]);
+    }
+
+    /**
+     * @param BasePageDocument $documentSource
+     * @param BasePageDocument|null $parentDocumentDestination
+     * @param string $localeDestination
+     */
+    protected function recursiveCopy(
+        BasePageDocument $documentSource,
+        BasePageDocument $parentDocumentDestination = null,
+        $localeDestination
+    ) {
+        // Generate new path.
+        $newPath = str_replace(
+            $this->sessionManager->getContentPath($this->webspaceKeySource),
+            $this->sessionManager->getContentPath($this->webspaceKeyDestination),
+            $documentSource->getPath()
+        );
+
+        $this->output->writeln('<info>Processing: </info>' . $documentSource->getPath() . ' => ' . $newPath);
+
+        /** @var BasePageDocument $documentDestination */
+        try {
+            $documentDestination = $this->documentManager->find($newPath, $localeDestination);
+        } catch (\Exception $exception) {
```

Comment:
Isn't there a better exception you can catch? What about the `DocumentNotFoundException`?